src/video_core/renderer_vulkan/liverpool_to_vk.cpp

```diff
@@ -1081,25 +1075,9 @@ vk::ClearValue ColorBufferClearValue(const AmdGpu::Liverpool::ColorBuffer& color
         break;
     }
 
-    if (num_components == 1) {
-        if (comp_swap != Liverpool::ColorBuffer::SwapMode::Standard) {
-            color.float32[static_cast<int>(comp_swap)] = color.float32[0];
-            color.float32[0] = 0.0f;
-        }
-    } else {
-        if (comp_swap_alt && num_components == 4) {
-            std::swap(color.float32[0], color.float32[2]);
-        }
-
-        if (comp_swap_reverse) {
-            std::reverse(std::begin(color.float32), std::begin(color.float32) + num_components);
-        }
-
-        if (comp_swap_alt && num_components != 4) {
-            color.float32[3] = color.float32[num_components - 1];
-            color.float32[num_components - 1] = 0.0f;
-        }
-    }
+    color.float32 = comp_swizzle.Apply(color.float32);
```

Comment:
VkClearColorValue is a union so you are technically applying the swizzle 3 times on same data

Reply:
Ah good catch, fixed.